addons/account/tests/test_account_move_in_invoice.py

```diff
@@ -2733,3 +2733,32 @@ def test_manual_label_change_on_payment_term_line(self):
         move_form.save()
         self.invoice.action_post()
         self.assertEqual(payment_term_line.name, 'XYZ', 'Manual name of payment term line should be kept')
+
+    def test_duplicate_invoice_with_separate_discount_acccount(self):
+        """When a separate discount account, make sure that discount lines don't have a tax_id set
+        So, when creating a credit notefrom the invoice, data are coherent (=same amount)"""
+        sale_tax = self.company_data['default_tax_sale']
+        self.company_data['company'].account_discount_expense_allocation_id = self.company_data['default_account_expense'].id
+        great_account = self.env['account.account'].create({
+            'name': '99999 Great Account',
+            'code': 'GA',
+            'account_type': 'income',
+            'tax_ids': [Command.set(sale_tax.ids)],
+            'company_id': self.company_data['company'].id,
+        })
+        invoice = self.init_invoice(move_type='out_invoice', amounts=[100], taxes=[sale_tax])
+        invoice.invoice_line_ids.write({
+            'discount': 10,
+            'account_id': great_account.id,
+        })
+        invoice.action_post()
+
+        credit_note_wizard = self.env['account.move.reversal'].with_context(
```

Comment:
To me, the credit note part is useless. Only check the invoice lines and the tax_ids and that's it

Reply:
Well, there is no test evaluating that a reversal gets the same amount as the original. I'd rather keep it to prevent further issues (although I do agree that for this specific case, it would be enough to do as you recommend)